src/calling/haplotypes/haplotypes.rs

```diff
@@ -55,10 +55,12 @@ impl AlleleFreqDist {
             let density = NotNan::new(y_0_prob).unwrap()
                 + (*vaf - *x_0) * (y_1_prob - y_0_prob) / (*x_1 - *x_0); //calculation of density for given vaf by linear interpolation
                                                                          //handle the case where density has 0 probability, which results in panicking with "FloatIsNan" error.
-            if density == NotNan::new(0.0).unwrap() {
-                Some(LogProb::ln_one())
-            } else {
+                                                                         //to avoid another floatisnan error, we have to be sure to return only probs that are certainly greater than 0.0 (0.000...)
+                                                                         //otherwise it should return zero probability, but in log space. (not 1 with ln_one() because density=0.0 means the probability is impossible.)
+            if density > NotNan::new(0.0).unwrap() {
                 Some(LogProb::from(Prob(NotNan::into_inner(density))))
+            } else {
+                Some(LogProb::ln_zero())
```

Comment:
The result of `LogProb::from(Prob(0.0))` should be the very same as `LogProb::ln_zero()`. If not, then that would be a bug with `LogProb::from`` I think.

Reply:
Yes, that might be true. Could it be the `fastexp()` that is said to potentially cause errors: https://docs.rs/bio/latest/i686-pc-windows-msvc/src/bio/stats/probs/mod.rs.html#414 ?

Reply:
But then we should just use LogProb::from here and don't have the if-else.

Reply:
We introduced it there in the previous PR: https://github.com/orthanq/orthanq/pull/21 when testing it on the HLA benchmark data. I'm pretty sure that we had a FloatIsNan error with a sample and it was only fixed with this line. However, I cannot reproduce that sample as the logs for Orthanq from previous runs do not exist.

Reply:
In this case I suggest that we remove the if-else block for now and merge the PR. If `FloatIsNan` appears back again, we come back to this PR. I quickly checked a few simulated and wastewater samples and there is no error with them with the removal of the if-else.

Reply:
ok!